### PR TITLE
ophcrack-cli: refactor to use ophcrack

### PR DESCRIPTION
**Pull request by Tochiaha**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Tochiaha** [comment]
@kirillrdy

**Comment by kirillrdy** [comment]
@Tochiaha can you change order of commits, otherwise ophcrack-cli is trying to access non existing attribute, which would break during something like git bisect

**Comment by Tochiaha** [comment]
@kirillrdy done.

**Comment by kirillrdy** [comment]
the order is still not right

at 9e6f504db61d27ea68c621811c902eacb5f2b388

if you try 
`nix-build -A ophcrack-cli`

```
       error: evaluation aborted with the following error message: 'lib.customisation.callPackageWith: Function called without required argument "ophcrack" at /home/kirillvr/nixpkgs/pkgs/by-name/op/ophcrack-cli/package.nix:1, did you mean "pkcrack"?'

```

**Comment by Tochiaha** [comment]
so the order of the commit should be just like yours correct?

**Comment by kirillrdy** [comment]
yes, the order is correct now, but commit message for ophcrack-cli should not be "init at"

**Comment by Tochiaha** [comment]
Thank you.

**Review by kirillrdy — APPROVED** [review]
Result of `nixpkgs-review pr 327030` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ophcrack</li>
    <li>ophcrack-cli</li>
  </ul>
</details>

appreciate you patience, we got there eventually !

File: pkgs/by-name/op/ophcrack-cli/package.nix [hunk 1]
```diff
@@ -1,73 +1 @@
...
-    description = "Free Windows password cracker based on rainbow tables";
-    homepage = "https://ophcrack.sourceforge.io";
-    license = with licenses; [ gpl2Plus ];
-    maintainers = with maintainers; [ tochiaha ];
-    mainProgram = "ophcrack-cli";
-    platforms = platforms.all;
-  };
-}
-
+{ ophcrack }: ophcrack.override { enableGui = false; }
```
Comment (Aleksanaa) [inline pkgs/by-name/op/ophcrack-cli/package.nix:1]: put `ophcrack-cli = ophcrack.override { enableGui = false; }` in all-packages.nix. Don't use this.

File: pkgs/by-name/op/ophcrack/package.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      if enableGui then
+        [
+          "--enable-gui"
+          "--with-qt5charts"
+        ]
+      else
+        [ "--disable-gui" ]
+    );
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/op/ophcrack/package.nix:58]: Better to avoid `with lib`.

File: pkgs/by-name/op/ophcrack/package.nix [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  expat,
+  enableGui ? true,
+}:
+stdenv.mkDerivation rec {
+
+  pname = "ophcrack";
+  version = "3.8.0";
+
+  src = fetchurl {
+    url = "https://sourceforge.net/projects/ophcrack/files/ophcrack/${version}/ophcrack-${version}.tar.bz2";
```
Comment (Aleksanaa) [inline pkgs/by-name/op/ophcrack/package.nix:18]:
```suggestion
    url = "mirror://ophcrack/ophcrack/${version}/ophcrack-${version}.tar.bz2";
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -11153,6 +11153,8 @@ with pkgs;
 
   ophis = python3Packages.callPackage ../development/compilers/ophis { };
 
+  ophcrack-cli = ophcrack.override { enableGui = false; };
```
Comment (Aleksanaa) [inline pkgs/top-level/all-packages.nix:11156]: So `c` is behind `i`... And how old is your local checkout?

Comment (Tochiaha) [reply pkgs/top-level/all-packages.nix:11156]: 5mins

**Comment by Aleksanaa** [comment]
You should still remove that `pkgs/by-name/op/opcrack-cli` directory

File: pkgs/by-name/op/ophcrack-cli/package.nix (hunk 1 above)
Comment (kirillrdy) [reply pkgs/by-name/op/ophcrack-cli/package.nix]: can't do because of by-name check

Comment (kirillrdy) [reply pkgs/by-name/op/ophcrack-cli/package.nix]: I guess by-name check allows to migrate from by-name, which is good for this case

**Comment by Aleksanaa** [comment]
Please reword commit message to `ophcrack-cli: refactor to use ophcrack` (not reactor)

**Comment by kirillrdy** [comment]
@Tochiaha thank you for your perseverance !

**Pull request merged**